vectorgrp/sil-kit
# docs: Improve demo and troubleshooting sections

## Changes

- Extended build instructions for demo, troubleshooting section
- Changed demo and utilitiy docs to rst description lists (instead of rst tables)
- Fixed various paragraphs and typos
- Centralized substitutions
- Restructured file structure to represent doc structure
- css for doxygenfunction

## Review Instructions

- Build the Docs
- Carefully read the sections with major changes:
  - Demos
  - Troubleshooting 
  - API (just the overview page)
- Review the root structure. Changed chapter titles: 
  - 'Simulation' -> 'Simulation Concetps'
  - 'API' -> 'C/C++ API'
- Condensed 'SIL Kit License' AND 'Third Party Licenses' on the root structure to just 'Licenses'
- Check the API sections that use breathe directives (doxygenfunction, doxygenclass, ...).
  The custom.css has selectors that effect all of them.
